libs/designer/src/lib/ui/panel/panelTabs/parametersTab/index.tsx

```diff
@@ -274,19 +274,42 @@ const ParameterSection = ({
     editorId: string,
     labelId: string,
     tokenPickerMode?: TokenPickerMode,
+    editorType?: string,
     isCodeEditor?: boolean,
     closeTokenPicker?: () => void,
     tokenPickerClicked?: (b: boolean) => void,
     tokenClickedCallback?: (token: ValueSegment) => void
   ): JSX.Element => {
-    const codeEditorFilteredTokens = tokenGroup.filter((group) => {
-      return group.id !== 'workflowparameters' && group.id !== 'variables';
-    });
+    const parameterType =
+      editorType ??
+      (nodeInputs.parameterGroups[group.id].parameters.find((param) => param.id === parameterId) ?? {})?.type ??
+      constants.SWAGGER.TYPE.ANY;
+    const supportedTypes: string[] = parameterType
+      ? getPropertyValue(constants.TOKENS, parameterType) || [constants.SWAGGER.TYPE.ANY, parameterType]
+      : null;
+
+    const filteredTokenGroup = tokenGroup.map((group) => ({
+      ...group,
+      tokens: group.tokens.filter((token: OutputToken) => {
+        if (isCodeEditor) {
+          return !(
+            token.outputInfo.functionName === constants.FUNCTION_NAME.VARIABLES ||
+            token.outputInfo.functionName === constants.FUNCTION_NAME.PARAMETERS ||
+            token.outputInfo.arrayDetails ||
+            token.key === constants.UNTIL_CURRENT_ITERATION_INDEX_KEY ||
+            token.key === constants.FOREACH_CURRENT_ITEM_KEY
+          );
+        }
+        return supportedTypes.some((supportedType) => equals(supportedType, token.type));
+      }),
+    }));
```

Comment:
Might not be needed, but if these have potential to take a while on large token counts we might want to wrap them in useMemos